src/core/style/filter_parameter.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2023 MapLibre contributors
+
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "filter_parameter.hpp"
+
+#include "style_parameter.hpp"
+
+namespace QMapLibre {
+
+/*!
+    \class FilterParameter
+    \brief A helper utility to manage filter parameters for a layer.
+    \ingroup QMapLibre
+
+    \headerfile filter_parameter.hpp <QMapLibre/FilterParameter>
+*/
+
+/*!
+    \brief Default constructor
+*/
+FilterParameter::FilterParameter(QObject *parent)
```

Comment:
Why does the implementation have documentation instead of the header?

Reply:
This is following Qt convention.

Reply:
OK